@here/harp-text-canvas/lib/rendering/TextMaterials.ts

```diff
@@ -158,84 +158,116 @@ const sdfTextFragmentSource: string = `
+     */
+    constructor(params?: RawShaderMaterialParameters) {
+        const isWebGL2 = params?.rendererCapabilities.isWebGL2 === true;
+
+        const shaderParams: THREE.ShaderMaterialParameters | undefined = params
+            ? {
+                  ...params,
+                  glslVersion: isWebGL2 ? THREE.GLSL3 : THREE.GLSL1,
+                  vertexShader:
+                      isWebGL2 && params.vertexShader
+                          ? convertVertexShaderToWebGL2(params.vertexShader)
+                          : params.vertexShader,
+                  fragmentShader:
+                      isWebGL2 && params.fragmentShader
+                          ? convertFragmentShaderToWebGL2(params.fragmentShader)
+                          : params.fragmentShader
+              }
+            : undefined;
+        // Remove properties that are not in THREE.ShaderMaterialParameters, otherwise THREE.js
+        // will log warnings.
+        if (shaderParams) {
+            delete (shaderParams as any).rendererCapabilities;
+        }
+        super(shaderParams);
     }
 }
 
 /**
  * @hidden
  * Material used for clearing glyphs from a [[GlyphTextureCache]].
  */
-export class GlyphClearMaterial extends THREE.RawShaderMaterial {
+export class GlyphClearMaterial extends RawShaderMaterial {
     /**
      * Creates a new `GlyphClearMaterial`.
-     *
+     * @param params - Material parameters. Always required except when cloning another
+     * material.
      * @returns New `GlyphClearMaterial`.
      */
-    constructor() {
-        const shaderParams: THREE.ShaderMaterialParameters = {
-            name: "GlyphClearMaterial",
-            vertexShader: clearVertexSource,
-            fragmentShader: clearFragmentSource,
-            uniforms: {},
-            depthTest: false,
-            depthWrite: false
-        };
+    constructor(params?: RendererMaterialParameters) {
+        const shaderParams: RawShaderMaterialParameters | undefined = params
+            ? {
+                  name: "GlyphClearMaterial",
+                  vertexShader: clearVertexSource,
+                  fragmentShader: clearFragmentSource,
+                  uniforms: {},
+                  depthTest: false,
+                  depthWrite: false,
+                  rendererCapabilities: params.rendererCapabilities
+              }
+            : undefined;
         super(shaderParams);
     }
-
-    // overrides with THREE.js base classes are not recognized by tslint.
-    onBeforeCompile(shader: THREE.Shader, renderer: THREE.WebGLRenderer) {
-        convertShader(shader, renderer);
-    }
 }
 
 /**
  * @hidden
  * Material used for copying glyphs into a [[GlyphTextureCache]].
  */
-export class GlyphCopyMaterial extends THREE.RawShaderMaterial {
+export class GlyphCopyMaterial extends RawShaderMaterial {
     /**
      * Creates a new `GlyphCopyMaterial`.
-     *
+     * @param params - Material parameters. Always required except when cloning another
```

Comment:
why are the params always required, if they are skipped, couldn't it just assume there is just GLSL v1 support?

Reply:
We want to support WebGL1 and WebGL2 transparently, so we need to be sure that every material gets the rendering capabilities at construction to know the webgl version. If the rendering capabilities are not passed to some materials, a webgl2 renderer will fail rendering them.